### PR TITLE
Add CLI argument to launch demo in a Platform

### PR DESCRIPTION
**Pull request by ottah**
Adds an extra cli arg `-p` or `--platform` that runs the demo application in a platform window. The demo applications manifest is written into a platform config, there isn't any types available so the type definition for the config is partially defined.
The provider window isn't placed into a platform window.

### CODE REVIEW

File: src/server/middleware.ts [hunk 1]
```diff
@@ -3,17 +3,19 @@ import * as path from 'path';
...
 
 /**
  * Creates express-compatible middleware function that will add/replace any URL's found within app.json files according
  * to the command-line options of this utility.
  */
-export function createAppJsonMiddleware(providerVersion: string, runtimeVersion?: string): RequestHandler {
+export function createAppJsonMiddleware(providerVersion: string, runtimeVersion?: string, platform: boolean = false): RequestHandler {
     return async (req: Request, res: Response, next: NextFunction) => {
         const configPath = req.params[0];            // app.json path, relative to 'res' dir
+        const isProvider = configPath.includes('provider.json');
```
Comment (pjbroadbent) [inline src/server/middleware.ts]: Provider manifest is typically `/provider/app.json`, which wouldn't be caught by this check. Should also consider how non-service applications will work here (seems it should be fine, though?).

Etiher `configPath.includes('provider')`, or reading the manifest and checking the UUID?..

File: src/utils/allowHook.ts [hunk 2]
```diff
@@ -34,7 +34,7 @@ export enum Hook {
 
 export interface HooksAPI {
     [Hook.DEFAULT_ARGS]: () => Partial<CLIArguments>;
-    [Hook.APP_MIDDLEWARE]: (app: express.Express) => void;
+    [Hook.APP_MIDDLEWARE]: (app: express.Express, args: Partial<CLIArguments>) => void;
```
Comment (pjbroadbent) [inline src/utils/allowHook.ts]: Why `Partial<CLIArguments>` over `CLIArguments`? The tooling should pass the parsed arguments here, where anything not specified on the CLI is filled-in with the default value.

File: src/server/middleware.ts [hunk 3]
```diff
@@ -29,9 +31,14 @@ export function createAppJsonMiddleware(providerVersion: string, runtimeVersion?
             config.startup_app.autoShow = true;
         }
 
+        let finalConfig: ManifestFile | PlatformManifest = config;
```
Comment (pjbroadbent) [inline src/server/middleware.ts]: Should there be an alias for this union? And rename `ManifestFile` to something more consistent with `PlatformManifest`?

Comment (ottah) [reply src/server/middleware.ts]: Created `Manifest` and renamed `ManifestFile` to `ClassicManifest`.

**Review by rypete — APPROVED** [review]

**Review by pjbroadbent — APPROVED** [review]

**Pull request merged**